src/http/web_socket/frame.cr

```diff
@@ -0,0 +1,196 @@
+abstract class HTTP::WebSocketFrame
```

Comment:
Yes, would do this eventually. As long as the current WebSocket class exists I can't do it.